src/map.cpp

```diff
@@ -120,7 +120,7 @@ void Map::setTile(uint16_t x, uint16_t y, uint8_t z, Tile* newTile)
 	uint32_t offsetX = x & FLOOR_MASK;
 	uint32_t offsetY = y & FLOOR_MASK;
 
-	Tile*& tile = floor->tiles[offsetX][offsetY];
+	std::shared_ptr<Tile>& tile = floor->tiles[offsetX][offsetY];
```

Comment:
Could this be `auto`
```suggestion
	auto& tile = floor->tiles[offsetX][offsetY];
```

Reply:
Man this code is VERY error prone hahahaha